sigle/src/modules/layout/components/AppHeader.tsx

```diff
@@ -31,6 +45,31 @@ export const AppHeader = () => {
   const { resolvedTheme, setTheme } = useTheme();
   const { user, isLegacy } = useAuth();
   const { status } = useSession();
+  const [showMobileHeaderDialog, setShowMobileHeaderDialog] = useState(false);
+  const [loadingCreate, setLoadingCreate] = useState(false);
+  const router = useRouter();
+
+  const handleCreateNewPrivateStory = async () => {
```

Comment:
we keep repeating this function everywhere, in the future we should think of a better way to do this, will be easy to forget some changes later on.

Reply:
yep that makes sense, should I create an issue for this so we don't forget?

Reply:
Yes please 👍 we could evaluate using smth like jotai or zustand maybe for this